tests/test_linux/test_ethtool.py

```diff
@@ -0,0 +1,28 @@
+import gc
+import os
+
+from pyroute2 import Ethtool
+
+
+def get_fds():
+    fd = os.open(f'/proc/{os.getpid()}/fd', os.O_RDONLY)
+    try:
+        return set(os.listdir(fd)) - {fd}
+    finally:
+        os.close(fd)
+
+
+def test_pipe_leak():
+    fds = get_fds()
+    etht = Ethtool()
+    etht.close()
```

Comment:
HUH. not calling `.close()` will leak descriptors anyway ?  You have to add `__del__` to classes that use `os.open()` / `os.pipe()` in class constructors.

Reply:
Personally I prefer resources to be freed explicitly, putting anyting in `__del__` requires a lot of checks and exception handling to guarantee no errors during the GC work.

But I'm to try to look what we can do there.

Reply:
@svinota just close the resources opened explicitly. 

```
from contextlib import suppress

class Xxx:
    def __init__(self):
            self.__x, self.__y = os.pipe()
            
     def close():
            # i.e. check every resource if it is closed already, close and clear the variable somehow.
            if self.__x !=1:
                os.close(self.__x)
                self.__x = -1 # not 'None' because of typing.
          
    def __del__(self):
            super().__del__() # is it required?
            self.close()

```

Reply:
Another way:
```python
class FdWrapper(int):
    def __del__(self):
        if not hasattr(self, '__closed'):
            try:
                print('here')
                os.close(self)
            except OSError:
                pass  # TODO: log ?
            finally:
                setattr(self, '__closed', True)


def MyPipe() -> tuple[FdWrapper, FdWrapper]:
    x, y = os.pipe()
    return FdWrapper(x), FdWrapper(y)


x, y = MyPipe()
os.write(y, b'qwe')

```